zip-0244.rst

```diff
@@ -1,8 +1,382 @@
 ::
 
   ZIP: 244
-  Title: Transaction Signature Validation for Transparent Zcash Extensions
+  Title: Transaction Non-Malleability
   Owners: Kris Nuttycombe <kris@electriccoin.co>
-  Status: Reserved
+          Daira Hopwood <daira@electriccoin.co>
+  Status: Draft
   Category: Consensus
-  Discussions-To: <https://github.com/zcash/zips/issues/384>
+  Created: 2021-01-06
+  License: MIT
+  Discussions-To: <https://github.com/zcash/zips/issues/411>
+
+===========
+Terminology
+===========
+
+The key words "MUST" and "MUST NOT" in this document are to be interpreted as described in RFC 2119. [#RFC2119]_
+
+The terms "consensus branch", "epoch", and "network upgrade" in this document are to be interpreted as
+described in ZIP 200. [#zip-0200]_
+
+========
+Abstract
+========
+
+This proposal defines a new transaction digest algorithm for the <TBD> network upgrade
+onward, in order to introduce non-malleable transaction identifiers that commit to
+all transaction data except for attestations to transaction validity.
+
+This proposal also defines a new transaction digest algorithm for signature validation,
+which shares all available structure produced during the construction of transaction 
+identifiers, in order to minimize redundant data hashing in validation.
+
+This proposal also defines new semantics for the ``hashLightClientRoot`` field of the
+block header, to enable additional commitments to be represented in this hash and to
+provide a mechanism for future extensibility of the set of commitments represented.
+
+==========
+Motivation
+==========
+
+In all cases, but particularly in order to support the use of transactions in
+higher-level protocols, any modification of the transaction that has not been
+explicitly permitted (such as via anyone-can-spend inputs) should invalidate
+attestations to spend authority or to the included outputs. Following the activation
+of this proposed change, transaction identifiers will be stable irrespective of 
+any possible malleation of witness data. 
```

Comment:
The "explicitly permitted" caveat seems to complicate the definition of when we can be sure that a txid is stable. I think it would be good to have a "nonmalleable" flag, such that if the flag is set then no features that allow malleation can be used.